### PR TITLE
fix: GitHub deprecation warnings

### PR DESCRIPTION
**Pull request by pau1ocampos**
Closes https://github.com/softprops/turnstyle/issues/43

## Context
GitHub will deprecate `set-output` command and `node12` as of 31st May 2023.
`@actions/core` should be upgraded for `v1.10.0`.
`action.yml` file should run on `node16`.
Please refer to this [documentation](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).

## Description

This pull request intends to upgrade `@actions/core` to `v1.10.0` and update `action.yml` file to contemplate `nodejs16` instead of `nodejs12` so that this action can be compliant with new GitHub requirements.

### CODE REVIEW

File: .node-version [hunk 1]
```diff
@@ -1 +1 @@
-12.14.1
\ No newline at end of file
+18.12.0
```
Comment (dhermes) [inline .node-version:1]: 16.18.0 is latest (as of 2022-11-03) in the 16.x series, wouldn't that make more sense given the target of `node16`?

Comment (pau1ocampos) [reply .node-version]: Good spot! Thank you! Changed it to `16.18.1` which is the newest stable version. Tested it and seems to be working fine. It's a pity that the project seems kind of abandoned. 😢

**Comment by michaelstepner** [comment]
FYI, Github has incorporated equivalent functionality to turnstyle into the Github Actions spec using the 'concurrency' parameter. softprops/turnstyle was developed in 2020 and Github [added the concurrency parameter in 2021](https://github.blog/changelog/2021-04-19-github-actions-limit-workflow-run-or-job-concurrency/).

The native 'concurrency' implementation has additional features beyond what's possible in turnstyle, and it is lower cost: it leaves jobs pending rather than running while they wait for a previous job to end.

**Comment by jeherve** [comment]
While the `concurrency` parameter is useful, I'm afraid there are still situations[^1] where it isn't enough to catch potential duplicated runs. In those situations, `softprops/turnstyle` can still be useful.

I would consequently recommend reopening this PR, it can still be useful in avoiding deprecation warnings for those still having to rely on `softprops/turnstyle`

Thanks!

[^1]: For reference, [here is one such case](https://github.com/Automattic/jetpack/issues/19099).

**Pull request closed without merging**